### PR TITLE
Create a disruption when we receive a realtime trip update

### PR DESCRIPTION
**Pull request by TeXitoi**
http://jira.canaltp.fr/browse/NAVP-270
http://jira.canaltp.fr/browse/NAVP-335

### CODE REVIEW

File: source/jormungandr/tests/disruptions_tests.py [hunk 1]
```diff
@@ -145,7 +147,7 @@ def test_disruption_with_stop_areas(self):
         when calling the pt object stopA, we should get its disruptions
         """
 
-        response = self.query_region('stop_areas/stopA?' + default_date_filter)
+        response = self.query_region('stop_areas/stopA?' + default_date_filter, display=True)
```
Comment (antoine-de) [inline source/jormungandr/tests/disruptions_tests.py]: you should remove the display=True flag

File: source/jormungandr/tests/disruptions_tests.py [hunk 2]
```diff
@@ -103,17 +103,19 @@ def test_disruption_end_point(self):
         eq_(stop_disrupt[0]['uri'], 'too_bad')
 
         """
-        by querying directly the impacted object, we only find active disruptions
+        by querying directly the impacted object, we only find publicable disruptions
```
Comment (xlqian) [inline source/jormungandr/tests/disruptions_tests.py]: publicable? Do you mean publishable? published?

File: source/kraken/fill_disruption_from_chaos.cpp [hunk 3]
```diff
@@ -132,6 +132,7 @@ make_severity(const chaos::Severity& chaos_severity, nt::new_disruption::Disrupt
 #undef EFFECT_ENUM_CONVERSION
     }
 
+    weak_severity = severity;
     return std::move(severity);
```
Comment (xlqian) [inline source/kraken/fill_disruption_from_chaos.cpp]: minor: `std::move` is not necessary  (I know it's not your code :) )

File: source/kraken/realtime.cpp [hunk 4]
```diff
@@ -57,7 +60,69 @@ static void cancel_vj(type::MetaVehicleJourney* meta_vj,
...
+    if (auto severity = weak_severity.lock()) { return std::move(severity); }
+
+    auto severity = boost::make_shared<nt::new_disruption::Severity>();
+    severity->uri = id;
+    severity->wording = "trip canceled!";
+    severity->created_at = timestamp;
+    severity->updated_at = timestamp;
+    severity->color = "#000000";
+    severity->priority = 42;
+    severity->effect = nt::new_disruption::Effect::NO_SERVICE;
```
Comment (xlqian) [inline source/kraken/realtime.cpp:78]: minor again:  Sorry for bothering, it's repeated code, but ok...

Comment (TeXitoi) [inline source/kraken/realtime.cpp:78]: If you have a good proposition to factorize this code easily, OK, but else, I let it like that.

File: source/kraken/tests/realtime_test.cpp [hunk 5]
```diff
@@ -39,6 +39,10 @@ www.navitia.io
 #include "routing/raptor.h"
 
 namespace nt = navitia::type;
+namespace pt = boost::posix_time;
+
+static const std::string feed_id = "42";
+static const pt::ptime timestamp = pt::ptime(boost::gregorian::date(2015,01,01),pt::time_duration(13,37,00));
```
Comment (antoine-de) [inline source/kraken/tests/realtime_test.cpp]: can't you use a "" operator ?

File: source/kraken/realtime.cpp (hunk 4 above)
Comment (xlqian) [inline source/kraken/realtime.cpp:78]: like I said, I'm Ok with that :)

**Comment by xlqian** [comment]
Ok for me

**Pull request merged**